pkg/client/client.go

```diff
@@ -0,0 +1,30 @@
+package client
+
+import (
+	"crypto/tls"
+
+	"golang.org/x/net/context"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+type tokenAuth struct {
+	token string
+}
+
+func (t *tokenAuth) GetRequestMetadata(context.Context, ...string) (map[string]string, error) {
+	return map[string]string{"token": t.token}, nil
+}
+
+func (*tokenAuth) RequireTransportSecurity() bool { return false }
+
+func New(cfg ClusterConfig) (*grpc.ClientConn, error) {
+	tlsConfig := &tls.Config{InsecureSkipVerify: true}
+	creds := credentials.NewTLS(tlsConfig)
+	return grpc.Dial(
+		cfg.Server,
+		grpc.WithPerRPCCredentials(&tokenAuth{cfg.Token}),
+		grpc.WithTransportCredentials(creds),
+	)
+}
```

Comment:
You're right... From [doc/comments](https://godoc.org/google.golang.org/grpc/credentials#PerRPCCredentials):
```
    // RequireTransportSecurity indicates whether the credentials requires
    // transport security
```
I made a dummy test with `RequireTransportSecury = true` on client without TLS transport and raise this error:
```
the credentials require transport level security (use grpc.WithTransportCredentials() to set)
```
I think we can change to `true` 👍